src/docs/testing/common-errors.md

```diff
@@ -0,0 +1,451 @@
+
+This Overlay widget cannot be marked as needing to build because the framework is already in the process of building widgets.  A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
+The widget on which setState() or markNeedsBuild() was called was: Overlay-[LabeledGlobalKey<OverlayState>#783d6]
+```
+
+### How might you run into the error?
+
+In general, this error occurs when the `setState` method is called within the build method. 
+
+A common scenario where this error would occur is when attempting to trigger a Dialog from within the build method. This scenario is often motivated by the requirement of showing some information to the user immediately upon their entry to a new page.  
+
+Below is a snippet that seems to be a common culprit of this error:
+
+
+```dart
+Widget build(BuildContext context) {
+
+  // Don't do this. 
+  showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text("Alert Dialog"),
+        );
+      });
+
+  return Center(
+    child: Column(
+      children: <Widget>[
+        Text('Show Material Dialog'),
+      ],
+    ),
+  );
+}
+```
+
+
+You can’t see the call to `setState` here, but it’s being used as part of `showDialog`, which is why this error is coming up. The build method is never the right place to call `showDialog`. The `build` method could be called by the framework for every frame, e.g., during an animation. 
+
+### How to fix it?
+
+One way to avoid this error is to use the Navigator API to trigger the dialog as a route. In the example below, there are two pages. The second page has a dialog to be displayed upon entry. When the user requests the second page from clicking on a button on the first page, the Navigator would push two routes in a row – one for the second page and another for the dialog.  
+
+
+```dart
+class FirstScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('First Screen'),
+      ),
+      body: Center(
+        child: RaisedButton(
+          child: Text('Launch screen'),
+          onPressed: () {
+            // Navigate to the second screen using a named route.
+            Navigator.pushNamed(context, '/second');
+            // Immediately show a dialog upon loading the second screen.
+            Navigator.push(
+              context,
+              PageRouteBuilder(
+                barrierDismissible: true,
+                opaque: false,
+                pageBuilder: (_, anim1, anim2) => MyDialog(),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+```
+
+## References
+
```

Comment:
Add a line. Like "To learn more blah blah, see ..."

Reply:
Done